Zend/zend_execute_API.c

```diff
@@ -596,6 +596,11 @@ ZEND_API int zval_update_constant_ex(zval *p, zend_class_entry *scope) /* {{{ */
 			if (UNEXPECTED(zend_ast_evaluate(&tmp, ast, scope) != SUCCESS)) {
 				return FAILURE;
 			}
+
+			if (UNEXPECTED(Z_ISERROR(tmp))) {
+				ZVAL_NULL(&tmp);
+			}
```

Comment:
This should no longer be necessary (due to the PROPAGATE_ERROR flag).

Reply:
Without this, test 7 fails.

```
const FOO=null;
const BAR=FOO[0];
var_dump(BAR);
```
results with an UNKNOWN:0 after printing the correct warning.